src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.DeclarationInfo.cs

```diff
@@ -64,6 +66,23 @@ internal struct NameDeclarationInfo
                 return default;
             }
 
+            private static bool IsTupleElementDeclaration(SyntaxToken token, SemanticModel semanticModel, int position, CancellationToken cancellationToken, out NameDeclarationInfo result)
+            {
+                result = IsLastTokenOfType<TupleElementSyntax>(token, semanticModel, tupleElement => tupleElement.Type, _ => default(SyntaxTokenList), _ => ImmutableArray.Create(SymbolKind.Local), cancellationToken);
+                return result.Type != null;
+            }
+
+            private static bool IsTupleExpressionDeclaration(SyntaxToken token, SemanticModel semanticModel, int position, CancellationToken cancellationToken, out NameDeclarationInfo result)
+            {
+                if (token.GetAncestor<TupleExpressionSyntax>() != null)
```

Comment:
❓ Why is this `GetAncestor`? I'm concerned that in a nested expression scenario this could provide incorrect results.

Reply:
The `TupleExpressionSyntax` is a few steps above the token:

```
(System.Array array, System.Action $$

IdentifierNameSyntax IdentifierName Action
↳ MemberAccessExpressionSyntax SimpleMemberAccessExpression System.Action
   ↳ ArgumentSyntax Argument System.Action
     ↳ TupleExpressionSyntax TupleExpression (System.Array array, System.Action
```

The call `IsFollowingTypeOrComma` does some further checks of the relationship between the token and the node so nesting shouldn't be a problem.